src/plugins/intel_cpu/src/nodes/mkldnn_rnn.cpp

```diff
@@ -772,20 +825,36 @@ void MKLDNNRNN::prepareParams() {
     }
 
     fillDescs();
-    if (cell_type == mkldnn::algorithm::vanilla_rnn) {
-        std::shared_ptr<vanilla_rnn_forward::desc> desc = descs[0];
-        prim.reset(new vanilla_rnn_forward(vanilla_rnn_forward::primitive_desc(*desc, getEngine())));
-    } else if (cell_type == mkldnn::algorithm::vanilla_gru) {
-        std::shared_ptr<gru_forward::desc> desc = descs[0];
-        prim.reset(new gru_forward(gru_forward::primitive_desc(*desc, getEngine())));
-    } else if (cell_type == mkldnn::algorithm::lbr_gru) {
-        std::shared_ptr<lbr_gru_forward::desc> desc = descs[0];
-        prim.reset(new lbr_gru_forward(lbr_gru_forward::primitive_desc(*desc, getEngine())));
-    } else if (cell_type == mkldnn::algorithm::vanilla_lstm) {
-        std::shared_ptr<lstm_forward::desc> desc = descs[0];
-        prim.reset(new lstm_forward(lstm_forward::primitive_desc(*desc, getEngine())));
+
+    RNNKey key = { inDataDescs, outDataDescs, wDescs, cell_type };
+
+    auto builder = [this](const RNNKey& key) -> std::shared_ptr<mkldnn::primitive> {
```

Comment:
Can we move `fillDescs` to the builder? It seems that this stage could be skipped if we can get the primitive from the cache. Moreover, it seems more logical to extract a function from the `fillDescs()` with `MKLDNNDescriptor createDesc()`  signature to reuse it in the builder without touch the descs vector at all.

Reply:
fillDescs fn was moved.
About descs vector. I see no reason to keep first desc instance there.